docker/Dockerfile.tests

```diff
@@ -8,28 +8,34 @@ FROM python:${PYTHON_VERSION} AS base
 RUN apt update && apt install default-jre unixodbc unixodbc-dev default-mysql-client locales locales-all -y && apt clean
 RUN python -m pip install --no-cache-dir --upgrade pip
 
-FROM base as build_local
-WORKDIR /great_expectations
-COPY . .
-
 FROM base AS build_github
 # BRANCH is the branch name you want to build
 ARG BRANCH=develop
 # We use --filter=tree:0 instead of --depth 1 because we really on git to get version information
 RUN git clone --filter=tree:0 --branch ${BRANCH} https://github.com/great-expectations/great_expectations.git
 WORKDIR /great_expectations
 
+FROM base as build_local
+WORKDIR /great_expectations
+COPY ./requirements.txt ./requirements.txt
```

Comment:
I believe you can copy multiple files all at once like this:
`COPY ./requirements.txt ./requirements-dev.txt ./constraints-dev.txt ./reqs ./`
You might need to break out `reqs` into it's own line because it is a directory.

The reason we want to put it all one 1 line is because docker will make a new layer for each line that makes a filesystem modification. This makes the docker image a little bit bigger.

Also, it doesn't matter if this block comes before or after the `FROM base AS build_github` block since either one or the other will build but not both.

Note, this change means we don't copy the whole local directory to the docker image, just the requirement files, but that is likely fine for local development, since you will be mounting your own directory.

Reply:
@billdirks I made the changes you described, thanks for the explanation.

> Note, this change means we don't copy the whole local directory to the docker image, just the requirement files, but that is likely fine for local development, since you will be mounting your own directory.

We are still doing `COPY . .` in the subsequent steps, so I think this would only be true if you use `--target dev`, correct?